File: PROGRAMAS/4_CALCULO_DE_PI/calculo_de_pi.rb

a = 0 # intervalo
b = 1 # intervalo
n = 100_000 # tamaño de la partición.
# h = (b-a) / n # fineza de la partición.
h = 0.00001
format('%<num>0.6f', num: h)
suma = 0 # aproximación de la integral
format('%<num>0.6f', num: suma)

(0..(n-1)).each {|i| suma += Math.sqrt(1-((a+i*h) * (a+i*h)))}

suma = h*((1-0) /2 + suma)

puts 4*suma

File: PROGRAMAS/4_CALCULO_DE_PI/calculo_de_pi.c

#include<stdio.h>
#include<math.h>

int main(){
	long double a = 0L, b = 1L; // intervalo.
	long double n = 100000L; // tamaño de la partición.
	long double h = (b-a)/n; // fineza de la partición.
	long double suma = 0L; // aproximación de la integral.

	for (int i=0;i<n-1;i++) suma += sqrtl(1 - (a+i*h) * (a+i*h));

	suma = h*((1-0) /2 + suma);
	printf("\n\n%.15Lf\n\n" , 4*suma);
	return 0;
}
